### PR TITLE
Add notification loading fix

### PR DESCRIPTION
**Pull request by milindk8**
# Description

While configuring new notification on global configuration it takes lots of time to load the data since we have lots of applications and pipelines. This should load as an async process like you are doing for others filters.

Fixes - https://github.com/devtron-labs/devtron/issues/1719

Screenshot 2022-05-19 at 10 24 10 AM

# Motivation

Taking a lot of time to load applications, environment & projects. So, loading data on user input will make loading faster & make UX better than the current implementation.

# Proposal

Load data when user give input directly through API

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
All tests have been performed manually.

- [x] By loading appList on dropdown of add notification

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/notifications/AddNotification.tsx [hunk 1]
```diff
@@ -55,20 +73,22 @@ interface AddNotificationState {
...
+        { value: 2, label: "project", type: "main" },
+        { value: 3, label: "environment", type: "main" },
     ];
     filterOptionsInner = [];
 
     constructor(props) {
         super(props);
         this.state = {
-            view: ViewType.LOADING,
+            view: ViewType.FORM,
```
Comment (vivek-devtron) [inline src/components/notifications/AddNotification.tsx:91]: Why are we changing it from loading to form?

File: src/components/notifications/AddNotification.tsx [hunk 2]
```diff
@@ -112,6 +134,10 @@ export class AddNotification extends Component<AddNotificationsProps, AddNotific
             filterInput: event.target.value,
             openSelectPipeline: true
         });
+        let unsavedFilter = this.state.appliedFilters.find(e => e.type && !e.value);
```
[the remark this answers is not in the document]
Comment (milindk8) [reply src/components/notifications/AddNotification.tsx:137]: we are using for the if condition
below it

File: src/components/notifications/notifications.css [hunk 3]
```diff
@@ -308,4 +308,11 @@
 
 .notification-tab__option .stroke-color {
     stroke: var(--N400);
-}
\ No newline at end of file
+}
+
+.select-button--pipeline-filter.loading-applist{
+    text-align: center;
```
Comment (vivek-devtron) [inline src/components/notifications/notifications.css:314]: can we minimize CSS?

File: src/components/notifications/AddNotification.tsx (hunk 1 above)
Comment (milindk8) [reply src/components/notifications/AddNotification.tsx]: done

**Review by vivek-devtron — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]
Looks Good

**Pull request merged**